src/Nest/Mapping/Types/Core/DateNanos/DateNanosProperty.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Diagnostics;
+using System.Runtime.Serialization;
+using Elasticsearch.Net.Utf8Json;
+
+namespace Nest
+{
+	[InterfaceDataContract]
+	public interface IDateNanosProperty : IDocValuesProperty
+	{
+		/// <summary>
+		/// Mapping field-level query time boosting. Accepts a floating point number, defaults to 1.0.
+		/// </summary>
+		[DataMember(Name = "boost")]
+		double? Boost { get; set; }
+
+		[DataMember(Name = "fielddata")]
+		INumericFielddata Fielddata { get; set; }
```

Comment:
Looking at the [`DateFieldMapper`](https://github.com/elastic/elasticsearch/blob/3400483af46189435e6039806dbee6b8fd9a8532/server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java), it doesn't look like this is supported. 

It looks like `INumericFielddata` should also be removed (via deprecated in 7.x), see https://github.com/elastic/elasticsearch/commit/e50eeeaffb33d7eaa0f7c2ebc71f1208460f4fcd#diff-abb176481762ceeebb86fa4c02c13993L79